src/main/scala/com.gu.gibbons/model/bonobo.scala

```diff
@@ -26,11 +29,15 @@ object User {
         id <- attrs.get("id").flatMap(a => Option(a.getS))
         email <- attrs.get("email").flatMap(a => Option(a.getS))
         name <- attrs.get("name").flatMap(a => Option(a.getS))
+        createdAt <- attrs.get("createdAt").flatMap(a => Option(a.getN).map(_.toLong))
```

Comment:
This is not your code, it was already like that, but it feels so frustrating to see this repeated code of `flatMap`ing over `Option()` and also using the raw DynamoDB API like `.getN` and `.getS`, when we already have Scanamo in place and it can do it for us.

If we introduced a simple little implict class here like this:

```scala
  implicit class RichAttrMap(av: AttributeValue) {
    def getAttribute[T : DynamoFormat](key: String): Option[T] =
      DynamoFormat[Map[String, T]].read(av).toOption.flatMap { attrs =>
        attrs.get(key)
      }
  }
```

then it would allow us to write this code in a much more readable way (to me at least):

```scala
    def read(av: AttributeValue) =
      (for {
        id <- av.getAttribute[String]("id")
        email <- av.getAttribute[String]("email")
        name <- av.getAttribute[String]("name")
        createdAt <- av.getAttribute[Long]("createdAt")
      } yield { ... }
```

I feel slightly confused as to why it was written in such a convoluted way when Regis is one of the contributors to Scanamo! Perhaps I am missing something.

Reply:
It really feels like much of this code is ripe for a refactor.